### DISCUSSION TITLE
Automatically detected the lowest Kubernetes version (but I don't have that version)

### DISCUSSION THREAD

**Discussion by cdemi** [siderolabs/talos#9487] [Q&A] [answered]
I am trying to upgrade Kubernetes from `v1.30.4` to `v1.31.1`. I have talos and talosctl of `v1.8.1`:

```
Client:
        Tag:         v1.8.1
        SHA:         477752fe
        Built:
        Go version:  go1.22.8
        OS/Arch:     windows/amd64
Server:
        NODE:        172.18.30.5
        Tag:         v1.8.1
        SHA:         477752fe
        Built:
        Go version:  go1.22.8
        OS/Arch:     linux/amd64
        Enabled:     RBAC
        NODE:        172.18.30.4
        Tag:         v1.8.1
        SHA:         477752fe
        Built:
        Go version:  go1.22.8
        OS/Arch:     linux/amd64
        Enabled:     RBAC
        NODE:        172.18.30.3
        Tag:         v1.8.1
        SHA:         477752fe
        Built:
        Go version:  go1.22.8
        OS/Arch:     linux/amd64
        Enabled:     RBAC
```

Kubernetes version is v1.30.4

```
NAME            STATUS   ROLES           AGE    VERSION
talos-45m-94p   Ready    control-plane   246d   v1.30.4
talos-e7g-8m1   Ready    control-plane   246d   v1.30.4
talos-v29-nuh   Ready    control-plane   246d   v1.30.4
```

When I try to upgrade with `talosctl upgrade-k8s --nodes 172.18.30.2  --to 1.31.1` I get:
```
automatically detected the lowest Kubernetes version 1.29.1
unsupported upgrade path 1.29->1.31 (from "1.29.1" to "1.31.1")
```

![image](https://github.com/user-attachments/assets/94aded73-1be9-4318-a324-90ffe1c02828)

I can't figure out why it's saying that I have `1.29.1` when all my nodes are `1.30.4`

Edit: I have updated the cluster to `1.30.5` but it still thinks I'm on 1.29.1

```
> talosctl upgrade-k8s --nodes 172.18.30.2  --to 1.31.1
automatically detected the lowest Kubernetes version 1.29.1
unsupported upgrade path 1.29->1.31 (from "1.29.1" to "1.31.1")

> k version  
Client Version: v1.31.1
Kustomize Version: v5.4.2
Server Version: v1.30.5
```

**Comment by rsmitty** [comment, marked as answer]
check your kube-proxy daemonset? I _think_ an upgrade should try to update that as well, but maybe it didn't?

**Reply by cdemi** [reply]
Thanks for the tip, you were indeed correct! The kube-proxy daemonset was stuck on v1.29.1. I upgraded it manually to v1.30.5 and I managed to start the upgrade to v1.31.1!

On the downside, after the upgrade to v1.31.1, the kube-proxy daemonset did not upgrade and remained on v1.30.5, so it looks like something might be wrong on my setup
